app/assets/javascripts/discourse/components/pwa-install-banner.js.es6

```diff
@@ -0,0 +1,64 @@
+import {
+  default as computed,
+  on
+} from "ember-addons/ember-computed-decorators";
+
+const USER_DISMISSED_PROMPT_KEY = "dismissed-pwa-install-banner";
+
+export default Ember.Component.extend({
+  deferredInstallPromptEvent: null,
+
+  _handleInstallPromptEvent(event) {
+    // Prevent Chrome 76+ from automatically showing the prompt
+    event.preventDefault();
+    // Stash the event so it can be triggered later
+    this.set("deferredInstallPromptEvent", event);
+  },
+
+  @on("init")
+  _registerListener() {
+    this._promptEventHandler = Ember.run.bind(
+      this,
+      this._handleInstallPromptEvent
+    );
+    window.addEventListener("beforeinstallprompt", this._promptEventHandler);
+  },
+
+  @on("willDestroyElement")
+  _unregisterListener() {
+    window.removeEventListener("beforeinstallprompt", this._promptEventHandler);
+  },
+
+  @computed
+  bannerDismissed: {
+    set(value) {
+      this.keyValueStore.set({ key: USER_DISMISSED_PROMPT_KEY, value: value });
+      return this.keyValueStore.get(USER_DISMISSED_PROMPT_KEY);
+    },
+    get() {
+      return this.keyValueStore.get(USER_DISMISSED_PROMPT_KEY);
+    }
+  },
+
+  @computed("deferredInstallPromptEvent", "bannerDismissed")
+  showPWAInstallBanner() {
+    return (
+      this.get("currentUser.trust_level") > 0 &&
+      this.deferredInstallPromptEvent && // Pass the browser engagement checks
+      !window.matchMedia("(display-mode: standalone)").matches && // Not be in the installed PWA already
+      !this.bannerDismissed // Have not a previously dismissed install banner
+    );
+  },
+
+  actions: {
+    turnOn() {
+      this.set("bannerDismissed", true);
+      this.deferredInstallPromptEvent.prompt();
+      this._unregisterListener();
```

Comment:
is there a risk this would be triggered again ? If not I would remove both calls and let willDestroy take care of this.